src/applications/bmqbrkr/bmqbrkr.m.cpp

```diff
@@ -592,50 +594,22 @@ run(bsl::ostream& errorDescription, TaskEnvironment* taskEnv, bool wait = true)
 
 int main(int argc, const char* argv[])
 {
-    printStartStopTrace("STARTING");
-
-    ignoreSigpipe();
-
-    // Register a SIGINT handler to allow graceful shutdown of the broker from
-    // lower layers.
-    struct sigaction sa;
-    sigemptyset(&sa.sa_mask);
-    sa.sa_flags   = 0;
-    sa.sa_handler = appShutdownSignal;
-
-    int rc = ::sigaction(SIGINT, &sa, NULL);
-    if (rc != 0) {
-        // Note that this is not fatal, as the primary way to stop the broker
-        // is not through a 'kill -s INT', but via issueing an 'exit' command
-        // through the control pipe.
-        bsl::cerr << "Failed to install SIGINT handler  (rc: " << rc << ")\n"
-                  << bsl::flush;
-    }
-
-    // If running in a TTY, enable signal handler so that user signals can be
-    // used to stop the task.
-    if (isatty(fileno(stdout))) {
-        ::sigaction(SIGQUIT, &sa, NULL);
-        ::sigaction(SIGTERM, &sa, NULL);
-    }
-
-    TaskEnvironment taskEnv;
-    s_taskEnv_p = &taskEnv;
 
     // Parse command line parameters
     bsl::string configDir;
     bsl::string instanceId = "default";
     bsl::string hostName;
     bsl::string hostTags;
     bsl::string hostDataCenter;
-    int         port = 0;
+    int         port    = 0;
+    bool        version = false;
 
     balcl::OptionInfo specTable[] = {
         {"",
          "config",
          "Path to the configuration directory",
          balcl::TypeInfo(&configDir),
-         balcl::OccurrenceInfo::e_REQUIRED},
+         balcl::OccurrenceInfo::e_OPTIONAL},
```

Comment:
Why is this changed to optional?  So that you can invoke `/path/to/bmqbrkr -v` to check its version, w/o the command line parser complaining about the config dir?

Reply:
Yes. The test for the `config` argument is done "manually" if `-v` is not passed.